### PR TITLE
add video2video

### PR DESCRIPTION
**Pull request by kangzhao2**

### CODE REVIEW

File: tests/pipelines/test_video2video.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import sys
+import unittest
+
+from modelscope.models import Model
+from modelscope.outputs import OutputKeys
+from modelscope.pipelines import pipeline
+from modelscope.utils.constant import DownloadMode, Tasks
```
Comment (XDUWQ) [inline tests/pipelines/test_video2video.py:7]: `DownloadMode` is not used

File: tests/pipelines/test_video2video.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+from modelscope.utils.test_utils import test_level
+
+
+class Video2VideoTest(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.task = Tasks.video_to_video
+        self.model_id = 'damo/Video-to-Video'
```
Comment (XDUWQ) [inline tests/pipelines/test_video2video.py:15]: `'damo/Video-to-Video'` seems to be unreachable, please make it publicly visible

File: modelscope/pipelines/multi_modal/video_to_video_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+class VideoToVideoPipeline(Pipeline):
+    r""" Video To Video Pipeline.
+
+    Examples:
+    >>> from modelscope.pipelines import pipeline
+    >>> from modelscope.outputs import OutputKeys
+
+    >>> p = pipeline('video-to-video', 'damo/Video-to-Video')
+    >>> input = {"video_path":VIDEOPATH, "text": TEXT}
+    >>> p(input,)
```
Comment (XDUWQ) [inline modelscope/pipelines/multi_modal/video_to_video_pipeline.py:33]: What does this mean?

File: modelscope/models/multi_modal/video_to_video/modules/autoencoder.py [hunk 4]
```diff
@@ -0,0 +1,587 @@
...
+        self.resolution = resolution
+        self.in_channels = in_channels
+        self.give_pre_end = give_pre_end
+        self.tanh_out = tanh_out
+
+        # compute block_in and curr_res at lowest res
+        block_in = ch * ch_mult[self.num_resolutions - 1]
+        curr_res = resolution // 2**(self.num_resolutions - 1)
+        self.z_shape = (1, z_channels, curr_res, curr_res)
+        logging.info('Working with z of shape {} = {} dimensions.'.format(
```
Comment (XDUWQ) [inline modelscope/models/multi_modal/video_to_video/modules/autoencoder.py:365]: please use modelscope logger

File: modelscope/models/multi_modal/video_to_video/modules/autoencoder.py [hunk 5]
```diff
@@ -0,0 +1,587 @@
...
+    def init_from_ckpt(self, path, ignore_keys=list()):
+        sd = torch.load(path, map_location='cpu')['state_dict']
+        keys = list(sd.keys())
+        sd_new = collections.OrderedDict()
+        for k in keys:
+            if k.find('first_stage_model') >= 0:
+                k_new = k.split('first_stage_model.')[-1]
+                sd_new[k_new] = sd[k]
+        self.load_state_dict(sd_new, strict=True)
+        logging.info(f'Restored from {path}')
```
Comment (XDUWQ) [inline modelscope/models/multi_modal/video_to_video/modules/autoencoder.py:495]: please modelscope logger

File: modelscope/models/multi_modal/video_to_video/modules/embedder.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
+# Copyright (c) Alibaba, Inc. and its affiliates.
+
+import logging
```
Comment (XDUWQ) [inline modelscope/models/multi_modal/video_to_video/modules/embedder.py:3]: `logging` had not been used

File: tests/pipelines/test_video2video.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        self.path = 'https://video-generation-wulanchabu.oss-cn-wulanchabu.aliyuncs.com/baishao/test.mp4'
+
+    @unittest.skipUnless(test_level() >= 1, 'skip test in current test level')
```
Comment (XDUWQ) [inline tests/pipelines/test_video2video.py:18]: Do you need to rely on `xformers` ? If not, please try not to skip unittest.

Comment (kangzhao2) [reply tests/pipelines/test_video2video.py:18]: Yes, we rely on xformers

File: modelscope/pipelines/multi_modal/video_to_video_pipeline.py [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+        vid_path = input['video_path']
+        if 'text' in input.keys():
+            text = input['text']
+        else:
+            text = ''
+
+        caption = text + self.model.positive_prompt
+        y = self.model.clip_encoder(caption).detach()
+
+        max_frames = 8
```
Comment (wenmengzhou) [inline modelscope/pipelines/multi_modal/video_to_video_pipeline.py:61]: 不要hardcode这个参数，通过preprocess_params传入， or pipeline kwargs传入

File: modelscope/pipelines/multi_modal/video_to_video_pipeline.py [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+        vid_path = input['video_path']
+        if 'text' in input.keys():
+            text = input['text']
+        else:
+            text = ''
+
+        caption = text + self.model.positive_prompt
+        y = self.model.clip_encoder(caption).detach()
+
+        max_frames = 32
```
Comment (wenmengzhou) [inline modelscope/pipelines/multi_modal/video_to_video_pipeline.py:61]: 可以使用self.model.max_frames？

**Review by wenmengzhou — APPROVED** [review]

**Pull request merged**